### ISSUE TITLE
[Bug] MultiColumnList property ColumnHeader::set

### ISSUE THREAD

**Issue by Ident** [cegui/cegui#171]
**[Original report](https://bitbucket.org/cegui/cegui/issue/171) by Paul Turner (Bitbucket: [pturner](https://bitbucket.org/pturner), GitHub: [pturner](https://github.com/pturner)).**

----------------------------------------

For ref: http://www.cegui.org.uk/phpBB2/viewtopic.php?t=2973

Will affect all versions.  Fix should be pretty easy (as stated in thread)

**Reproducibility:** always

**Additional information:** Text quoted from forum thread:

---- Start ----
I'm pretty sure I've found a bug in ceguimulticolumnlistproperties.cpp

Here is the function, copied for convenience:

Code:
void ColumnHeader::set(PropertyReceiver* receiver, const String& value)
{
   // extract data from the value string

   size_t wstart = value.find("width:");
   size_t idstart = value.find("id:");

   String caption(value.substr(0, wstart));
   caption = caption.substr(caption.find_first_of(":") + 1);

        // *** [Edit:  Next line is bugged] ***
   String width(value.substr(wstart, idstart));
   width = width.substr(width.find_first_of(":") + 1);

   String id(value.substr(idstart));
   id = id.substr(id.find_first_of(":") + 1);

   static_cast<MultiColumnList*>(receiver)->addColumn(
      caption, PropertyHelper::stringToUint(id), PropertyHelper::stringToUDim(width));
}

Basically this function is trying to chop apart a string that looks something like this: 'text:Distance width:{0.25,0} id:3'

In the offending line, our substr params are wrong, and we're overflowing the buffer. The bounds checking inside CEGUI::String doesn't quite do the proper check to catch it.

The offending line should be changed to:
String width(value.substr(wstart, idstart-wstart));

And possibly the CEGUI::String bounds checking should consider how many characters are being copied, not just the initial position.

Thanks!
---- End ----
